Luciana Godio/javascript/index.js

```diff
@@ -0,0 +1,98 @@
+const array1 = [
+  { name: "Captain America" },
+  { name: "Hulk" },
+  { name: "Professor X" },
+  { name: "Storm" },
+  { name: "Spiderman" },
+];
+console.log("Marvel Characters: ", array1);
+
+const obj1 = {
+  1: { name: "Beast" },
+  2: { name: "Firestar" },
+  3: { name: "Professor X" },
+  4: { name: "Storm" },
+};
+
+const obj2 = {
+  1: { name: "Iron man" },
+  2: { name: "Hulk" },
+  3: { name: "Spiderman" },
+  4: { name: "Thor" },
+};
+
+// 1) Given an array of objects which contains character of Marvel, filter the characters that are members of X-Men.
+const filteredXmenCharacters = array1.filter((character) =>
+  Object.values(obj1)
+    .map((character) => character.name)
+    .includes(character.name)
+);
+console.log("Filtered X-Men Characters from Marvel: ", filteredXmenCharacters);
+
+// 2) Given 2 arrays of objects (obj1 and obj2) one with members o Avengers and one with members of X-Men, find the characters that are on both teams.
+const listCharactersForFilter = ["Beast", "Hulk", "Storm", "Thor", "Iron man"];
+console.log("List of characters to filter: ", listCharactersForFilter);
+
+const charactersInXmen = Object.values(obj1).map((character) => character.name);
+console.log("X-Men:", charactersInXmen);
+
+const charactersInAvengers = Object.values(obj2).map(
+  (character) => character.name
+);
+console.log("Avengers:", charactersInAvengers);
+
+const filterBothTeams = listCharactersForFilter.filter(
+  (character) =>
+    charactersInXmen.includes(character) &&
```

Comment:
La parte del and (&&) en el filter no me funciono. Si pongo uno u otro me funciona. Capaz se podria filtrar un solo array haciendo un spread de los 2 anteriorers (charactersInXmen y charactersInAvengers )